### PR TITLE
LESS: fix comments in value lists

### LINKED ISSUE #9354: LESS Regression: Comments in value lists
Both space- and comma-separated [value lists](http://lesscss.org/functions/#list-functions) in LESS are broken when they contain comments after the individual values. This regressed somewhere between Prettier 2.0.5 and 2.1.2.

**Prettier 2.1.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEABeBnGBaDAHAQzDlzkICcD4ATJAAgGIDmW6B6NugZRgPJjoB3AJYwAFnQCCAHSh1GAIyXL2ncQjoYIAWzh0AQrPkMwpswG5VdAlGp1yEAK626ouk4EjxdAMKzZ6HBY2JDa2gSkFFRwtIwsLAA0Vjx8nqISMnKKykpJHHTqclq6BkaMZhZWNnYOznZuHkLpviAJIBB4MMLQGMigfA6CAAp8CL0oBAA2ggQAnr1tCpRgANZwMFyEYMJQAObIMOSOcG1w2gox1DEAMja7jgS7cABiEOThMF17yCAEjjAQVogMQwbSTADqYlEQS2cC4Y1EwgAbqJZj8wBgFiAdhg4PwhpRduFkAAzKa4toAKwwAA99Ms1hsCLprjs4KTyScQNSaVwdrtJnAAIqOCDwDmTCkgCi48g-QWYoF4cg7GDg4TUcTIAAcAAY2sqILjwZQ8D9lUE8Uj2W0AI6i+AEjrjX4YbBQOAxGJA8hwe3CX0Ex7EpBkyVc3HaYQHI4R-mCkVi9mhzltXgKdWasTIABMacowkm-J8OhDICCAFYgY5cQAVAgKcZhqVI44ASVsCA2YBVnUkth4s0FEtxAF9R0A)
```sh
--parser less
```

**Input:**
```less
@test-space-separated: #aaaaaa // Start with A
  #bbbbbb // then some B
  #cccccc; // and round it out with C

@test-comma-separated: #aaaaaa, // Start with A
  #bbbbbb, // then some B
  #cccccc; // and round it out with C
```

**Output:**
```less
@test-space-separated: #aaaaaa a // Start with
  #bbbbbb b // then some
  #cccccc; // and round it out with C

@test-comma-separated: #aaaaaa,
  , // Start with
  #bbbbbb,
  , // then some
  #cccccc; // and round it out with C

```

**Expected behavior:**

The input is unchanged.

### PR DESCRIPTION
**Pull request by thorn0**
Fixes #9354 

- [x] I’ve added tests to confirm my change works.
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Review by fisker — APPROVED** [review]

**Review by evilebottnawi — APPROVED** [review]

File: src/language-css/loc.js [hunk 1]
```diff
@@ -69,18 +69,25 @@ function calculateValueNodeLoc(node, rootOffset, text) {
...
-      : 0) +
-    (node.type !== "css-atrule" &&
+  let result = node.source.startOffset;
+  if (typeof node.prop === "string") {
+    result += node.prop.length;
+  }
+
+  if (node.type === "css-atrule" && typeof node.name === "string") {
+    result +=
+      1 + node.name.length + node.raws.afterName.match(/^[\s:]*/)[0].length;
```
Comment (fisker) [inline src/language-css/loc.js:79]: Did you mean `/^[\s:]?/` ？

Comment (thorn0) [reply src/language-css/loc.js:79]: No. It calculates the length of the leading substring that consists of only colons and whitespace.
E.g. `\t\t:\t//foo` → 4.

Comment (thorn0) [reply src/language-css/loc.js]: I added this explanation as a comment in the code.

Comment (fisker) [reply src/language-css/loc.js]: Sorry, my main concern is about `:`, shouldn't only  one colon be  matched? So `/^(?:\s*:\s*)?/`?

Comment (thorn0) [reply src/language-css/loc.js]: If you know a way how more than one colon can end up there, then yes.

Comment (fisker) [reply src/language-css/loc.js]: I don't know, but if you mean to match one `:`, I think you should write in this way.

**Review by fisker — APPROVED** [review]

**Pull request merged**